File: examples/governance_static_go/domain/approve_review.go
package domain

import (
	"time"
	"fmt"
)

type ApproveReview struct {
	ReviewId string `json:"review_id"`
	Notes string `json:"notes"`
}

func (c ApproveReview) CommandName() string { return "ApproveReview" }

func (c ApproveReview) Execute(repo ComplianceReviewRepository) (*ComplianceReview, *ApprovedReview, error) {
	existing, err := repo.Find(c.ReviewId)
	if err != nil {
		return nil, nil, err
	}
	if existing == nil {
		return nil, nil, fmt.Errorf("ComplianceReview not found: %s", c.ReviewId)
	}
	existing.Notes = c.Notes
	if existing.Status != "open" && existing.Status != "changes_requested" {
		return nil, nil, fmt.Errorf("cannot ApproveReview: ComplianceReview is in %s state", existing.Status)
	}
	existing.Status = "approved"
	existing.UpdatedAt = time.Now()
	if err := existing.Validate(); err != nil {
		return nil, nil, err
	}
	if err := repo.Save(existing); err != nil {
		return nil, nil, err
	}
	event := ApprovedReview{
		AggregateID: existing.ID,
		ReviewId: c.ReviewId,
		Notes: c.Notes,
		OccurredAt: time.Now(),
	}
	return existing, &event, nil
}

File: examples/governance/compliance_domain/lib/compliance_domain/compliance_review/commands/approve_review.rb
module ComplianceDomain
  class ComplianceReview
    module Commands
      class ApproveReview
        include Hecks::Command
        emits "ApprovedReview"

        attr_reader :review_id, :notes

        def initialize(review_id: nil, notes: nil)
          @review_id = review_id
          @notes = notes
        end

        def call
          existing = repository.find(review_id)
          if existing
            unless ["open", "changes_requested"].include?(existing.status)
              raise ComplianceDomain::Error, "Cannot ApproveReview: status must be one of open, changes_requested, got '#{existing.status}'"
            end
            ComplianceReview.new(
              id: existing.id,
              model_id: existing.model_id,
              policy_id: existing.policy_id,
              reviewer_id: existing.reviewer_id,
              notes: notes,
              conditions: existing.conditions,
              outcome: "approved",
              completed_at: Time.now.to_s,
              status: "approved"
            )
          else
            raise ComplianceDomain::Error, "ComplianceReview not found: #{review_id}"
          end
        end
      end
    end
  end
end
